src/hub/ShareClassManager.sol

```diff
@@ -396,21 +357,23 @@ contract ShareClassManager is Auth, IShareClassManager {
             userOrder.lastUpdate,
             investor,
             depositAssetId,
-            depositAssetAmount,
+            paymentAssetAmount,
             userOrder.pending,
             payoutShareAmount,
             epochAmounts.issuedAt
         );
 
-        userOrder.lastUpdate += 1;
-        // If there is nothing to claim anymore we can short circuit the in between epochs
+        // If there is nothing to claim anymore we can short circuit to the latest epoch
         if (userOrder.pending == 0) {
             // The current epoch is always one step ahead of the stored one
             userOrder.lastUpdate = nowDepositEpoch(scId_, depositAssetId);
             canClaimAgain = false;
+        } else {
+            userOrder.lastUpdate += 1;
         }
 
-        if (depositEpochId[scId_][depositAssetId] == userOrder.lastUpdate) {
+        // If user claimed up to latest approval epoch, move queued to pending
+        if (userOrder.lastUpdate == nowDepositEpoch(scId_, depositAssetId)) {
```

Comment:
@GalloDaSballo : Fixes https://github.com/Recon-Fuzz/centrifuge-review-protocol-v3/issues/34

Essentially, the new and only `claimDeposit` is similar to the previous `claimDepositUntilEpoch`. However, we only move queued amounts to pending when all claims have been processed (represented by `userOrder.lastUpdate = now{Deposit, Redeem}Epoch`.

Also, please note that when moving queued to pending, we update the total pending as well (increment if queued is not cancelling, decrement if queued is cancelling).